src/CakeSlayer/Has.hs

```diff
@@ -56,3 +65,69 @@ class Has field env where
 grab :: forall field env m . (MonadReader env m, Has field env) => m field
 grab = asks $ obtain @field
 {-# INLINE grab #-}
+
+
+{- | The newtype for deriving via mechanism for 'Has' class. Let's see how
+it's supposed to be used.
+
+Imagine that we have a data type like this:
+
+@
+__data__ Env = Env
+    { envInt :: !'Int'
+    , envString :: !'String'
+    }
+@
+
+GHC generates the following 'HasField' instances automatically:
+
+@
+__instance__ 'HasField' "envInt" Env 'Int' __where__
+    'getField' :: Env -> 'Int'
+    'getField' = envInt
+
+__instance__ 'HasField' "envString" Env 'String' __where__
+    'getField' :: Env -> 'String'
+    'getField' = envString
+@
+
+You can have now see that using @'getField' \@\"envInt\"@ is equivalent to @envInt@.
+
+So, that means that it's possible to improve 'Has' instances interface and remove some boilerplate.
+
+Instead of writing the following code:
+
+@
+__data__ Env = Env
+    { envInt :: !'Int'
+    , envString :: !'String'
+    }
+
+__instance__ 'Has' 'Int' Env __where__
+    'obtain' = envInt
+    {-# INLINE 'obtain' #-}
+
+__instance__ 'Has' 'String' Env __where__
+    'obtain' = envString
+    {-# INLINE 'obtain' #-}
+@
+
+It should be possible to write it using `DerivingVia` extension:
+
+@
+data Env = Env
+    { envInt :: !Int
+    , envString :: !String
+    } deriving (Has Int) via Field "envInt" Int Env
+      deriving (Has String) via Field "envString" String Env
```

Comment:
Is it good idea to omit `()` here and also use bold for `deriving` and `via` words? :thinking: 
```suggestion
      deriving Has String via Field "envString" String Env
```

Reply:
It doesn't parse without parenthesis...

Reply:
Ah, I didn't know that!